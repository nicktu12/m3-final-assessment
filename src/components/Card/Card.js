import React from 'react';

const Card = () => {
  return(
    <div>
      Hey!!!
    </div>
  )
}

export default Card;
