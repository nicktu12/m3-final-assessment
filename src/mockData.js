export const mockHouses = [{"url":"https://www.anapioficeandfire.com/api/houses/1","name":"House Algood","region":"The Westerlands","coatOfArms":"A golden wreath, on a blue field with a gold border(Azure, a garland of laurel within a bordure or)","words":"","titles":[""],"seats":[""],"currentLord":"","heir":"","overlord":"https://www.anapioficeandfire.com/api/houses/229","founded":"","founder":"","diedOut":"","ancestralWeapons":[""],"cadetBranches":[],"swornMembers":[]},{"url":"https://www.anapioficeandfire.com/api/houses/2","name":"House Allyrion of Godsgrace","region":"Dorne","coatOfArms":"Gyronny Gules and Sable, a hand couped Or","words":"No Foe May Pass","titles":[""],"seats":["Godsgrace"],"currentLord":"https://www.anapioficeandfire.com/api/characters/298","heir":"https://www.anapioficeandfire.com/api/characters/1922","overlord":"https://www.anapioficeandfire.com/api/houses/285","founded":"","founder":"","diedOut":"","ancestralWeapons":[""],"cadetBranches":[],"swornMembers":["https://www.anapioficeandfire.com/api/characters/298","https://www.anapioficeandfire.com/api/characters/1129","https://www.anapioficeandfire.com/api/characters/1301","https://www.anapioficeandfire.com/api/characters/1922"]},{"url":"https://www.anapioficeandfire.com/api/houses/3","name":"House Amber","region":"The North","coatOfArms":"","words":"","titles":[""],"seats":[""],"currentLord":"","heir":"","overlord":"","founded":"","founder":"","diedOut":"","ancestralWeapons":[""],"cadetBranches":[],"swornMembers":[]},{"url":"https://www.anapioficeandfire.com/api/houses/4","name":"House Ambrose","region":"The Reach","coatOfArms":"Or, semy of ants gules","words":"Never Resting","titles":[""],"seats":[""],"currentLord":"https://www.anapioficeandfire.com/api/characters/141","heir":"","overlord":"https://www.anapioficeandfire.com/api/houses/398","founded":"","founder":"","diedOut":"","ancestralWeapons":[""],"cadetBranches":[],"swornMembers":["https://www.anapioficeandfire.com/api/characters/82","https://www.anapioficeandfire.com/api/characters/102","https://www.anapioficeandfire.com/api/characters/141","https://www.anapioficeandfire.com/api/characters/152","https://www.anapioficeandfire.com/api/characters/344"]}]
