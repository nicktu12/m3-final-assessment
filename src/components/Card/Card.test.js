import React from 'react';
import { shallow } from 'enzyme';
import Card from './Card';
import { mockHouses } from '../../mockData';

describe('card', ()=>{
  it('should match snapshot', ()=> {
    const wrapper = shallow(
      <Card houseInfo={mockHouses[0]} key={1} />
    )
    expect(wrapper).toMatchSnapshot();
  })
})
