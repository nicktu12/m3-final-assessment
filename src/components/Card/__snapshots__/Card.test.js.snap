// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`card should match snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Card
    houseInfo={
        Object {
            "ancestralWeapons": Array [
              "",
            ],
            "cadetBranches": Array [],
            "coatOfArms": "A golden wreath, on a blue field with a gold border(Azure, a garland of laurel within a bordure or)",
            "currentLord": "",
            "diedOut": "",
            "founded": "",
            "founder": "",
            "heir": "",
            "name": "House Algood",
            "overlord": "https://www.anapioficeandfire.com/api/houses/229",
            "region": "The Westerlands",
            "seats": Array [
              "",
            ],
            "swornMembers": Array [],
            "titles": Array [
              "",
            ],
            "url": "https://www.anapioficeandfire.com/api/houses/1",
            "words": "",
          }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": null,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h3>
          House Algood
</h3>,
        "",
        <p>
          <span>
                    House Seat:
          </span>
           
          
</p>,
        <p>
          <span>
                    Titles:
          </span>
           
          
</p>,
        <p>
          <span>
                    Coat of Arms:
          </span>
           
          A golden wreath, on a blue field with a gold border(Azure, a garland of laurel within a bordure or)
</p>,
        <p>
          <span>
                    Ancestral Weapons:
          </span>
           
          
</p>,
        "",
      ],
      "className": "cards",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": null,
        "nodeType": "host",
        "props": Object {
          "children": "House Algood",
        },
        "ref": null,
        "rendered": "House Algood",
        "type": "h3",
      },
      "",
      Object {
        "instance": null,
        "key": null,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <span>
              House Seat:
</span>,
            " ",
            "",
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": null,
            "nodeType": "host",
            "props": Object {
              "children": "House Seat:",
            },
            "ref": null,
            "rendered": "House Seat:",
            "type": "span",
          },
          " ",
          "",
        ],
        "type": "p",
      },
      Object {
        "instance": null,
        "key": null,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <span>
              Titles:
</span>,
            " ",
            "",
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": null,
            "nodeType": "host",
            "props": Object {
              "children": "Titles:",
            },
            "ref": null,
            "rendered": "Titles:",
            "type": "span",
          },
          " ",
          "",
        ],
        "type": "p",
      },
      Object {
        "instance": null,
        "key": null,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <span>
              Coat of Arms:
</span>,
            " ",
            "A golden wreath, on a blue field with a gold border(Azure, a garland of laurel within a bordure or)",
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": null,
            "nodeType": "host",
            "props": Object {
              "children": "Coat of Arms:",
            },
            "ref": null,
            "rendered": "Coat of Arms:",
            "type": "span",
          },
          " ",
          "A golden wreath, on a blue field with a gold border(Azure, a garland of laurel within a bordure or)",
        ],
        "type": "p",
      },
      Object {
        "instance": null,
        "key": null,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <span>
              Ancestral Weapons:
</span>,
            " ",
            "",
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": null,
            "nodeType": "host",
            "props": Object {
              "children": "Ancestral Weapons:",
            },
            "ref": null,
            "rendered": "Ancestral Weapons:",
            "type": "span",
          },
          " ",
          "",
        ],
        "type": "p",
      },
      "",
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": null,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h3>
            House Algood
</h3>,
          "",
          <p>
            <span>
                        House Seat:
            </span>
             
            
</p>,
          <p>
            <span>
                        Titles:
            </span>
             
            
</p>,
          <p>
            <span>
                        Coat of Arms:
            </span>
             
            A golden wreath, on a blue field with a gold border(Azure, a garland of laurel within a bordure or)
</p>,
          <p>
            <span>
                        Ancestral Weapons:
            </span>
             
            
</p>,
          "",
        ],
        "className": "cards",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": null,
          "nodeType": "host",
          "props": Object {
            "children": "House Algood",
          },
          "ref": null,
          "rendered": "House Algood",
          "type": "h3",
        },
        "",
        Object {
          "instance": null,
          "key": null,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <span>
                House Seat:
</span>,
              " ",
              "",
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": null,
              "nodeType": "host",
              "props": Object {
                "children": "House Seat:",
              },
              "ref": null,
              "rendered": "House Seat:",
              "type": "span",
            },
            " ",
            "",
          ],
          "type": "p",
        },
        Object {
          "instance": null,
          "key": null,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <span>
                Titles:
</span>,
              " ",
              "",
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": null,
              "nodeType": "host",
              "props": Object {
                "children": "Titles:",
              },
              "ref": null,
              "rendered": "Titles:",
              "type": "span",
            },
            " ",
            "",
          ],
          "type": "p",
        },
        Object {
          "instance": null,
          "key": null,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <span>
                Coat of Arms:
</span>,
              " ",
              "A golden wreath, on a blue field with a gold border(Azure, a garland of laurel within a bordure or)",
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": null,
              "nodeType": "host",
              "props": Object {
                "children": "Coat of Arms:",
              },
              "ref": null,
              "rendered": "Coat of Arms:",
              "type": "span",
            },
            " ",
            "A golden wreath, on a blue field with a gold border(Azure, a garland of laurel within a bordure or)",
          ],
          "type": "p",
        },
        Object {
          "instance": null,
          "key": null,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <span>
                Ancestral Weapons:
</span>,
              " ",
              "",
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": null,
              "nodeType": "host",
              "props": Object {
                "children": "Ancestral Weapons:",
              },
              "ref": null,
              "rendered": "Ancestral Weapons:",
              "type": "span",
            },
            " ",
            "",
          ],
          "type": "p",
        },
        "",
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactFifteenAdapter {
      "options": Object {
        "supportPrevContextArgumentOfComponentDidUpdate": true,
      },
    },
  },
}
`;
